includes/AI/RestController.php

```diff
@@ -0,0 +1,1312 @@
+<?php
+
+namespace WeDevs\Wpuf\AI;
+
+use WP_REST_Server;
+use WP_REST_Request;
+use WP_REST_Response;
+use WP_Error;
+
+/**
+ * REST API Controller for AI Form Builder
+ * 
+ * Handles all REST API endpoints for AI form generation with comprehensive
+ * security, validation, and error handling. Implements rate limiting,
+ * XSS protection, and proper WordPress capability checks.
+ * 
+ * Security Features:
+ * - Rate limiting (10 requests/hour per user)
+ * - XSS protection with field sanitization
+ * - Proper capability checks with role filtering
+ * - Session validation and timeout handling
+ * - Enhanced error logging with context
+ * 
+ * @since 1.0.0
+ * @version 1.2.0
+ */
+class RestController {
```

Comment:
always extend `WP_REST_Controller` class for creating WordPress Rest API. see existing file structure at `includes/API.php` and `includes/Api/**`